packages/deployment-service/src/resolve-production-s3-bucket-permissions.ts

```diff
@@ -0,0 +1,132 @@
+import {
+  PutPublicAccessBlockCommand,
+  S3Client,
+  BucketCannedACL,
+  PutBucketAclCommand,
+  PutBucketOwnershipControlsCommand,
+  DeleteBucketPolicyCommand,
+  PutBucketPolicyCommand,
+} from '@aws-sdk/client-s3'
+import { OnEventHandler } from 'aws-cdk-lib/custom-resources/lib/provider-framework/types'
+
+const bucketArn = (bucketName: string): string => `arn:aws:s3:::${bucketName}`
+
+type BucketPolicyStatement = {
+  Effect: 'Allow' | 'Deny'
+  Action: string[]
+  Condition?: string[]
+  Resource: string
+  Principal: {
+    [s: string]: string
+  }
+}
+
+const resolveBucket = async (client: S3Client, bucketName: string) => {
+  await client.send(
+    new DeleteBucketPolicyCommand({
+      Bucket: bucketName,
+    }),
+  )
+
+  await client.send(
+    new PutBucketAclCommand({
+      Bucket: bucketName,
+      ACL: BucketCannedACL.private,
+    }),
+  )
+
+  await client.send(
+    new PutBucketOwnershipControlsCommand({
+      Bucket: bucketName,
+      OwnershipControls: {
+        Rules: [
+          {
+            ObjectOwnership: 'BucketOwnerEnforced',
+          },
+        ],
+      },
+    }),
+  )
+
+  const statements: BucketPolicyStatement[] = [
+    {
+      Effect: 'Allow',
+      Action: ['s3:Get*', 's3:List*', 's3:Put*'],
+      Resource: `${bucketArn(bucketName)}/*`,
+      Principal: {
+        AWS: '*',
+      },
+      // Condition?? TODO for PaaS account
```

Comment:
were we wanting to do this now?

Reply:
I decided to do these changes in another PR once this has all been sorted. Wasn't confident with the conditionals as there's no types